app/src/main/java/woowacourse/movie/movie/adapter/MovieAdapter.kt

```diff
@@ -0,0 +1,53 @@
+package woowacourse.movie.movie.adapter
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import woowacourse.movie.R
+import woowacourse.movie.ui.model.MovieFeedUiModel
+
+class MovieAdapter(
+    private val items: List<MovieFeedUiModel>,
+    private val onReserveClick: ReserveClickListener,
+) : RecyclerView.Adapter<ViewHolder>() {
+    override fun getItemCount(): Int = items.size
+
+    override fun getItemViewType(position: Int): Int {
+        return when (items[position]) {
+            is MovieFeedUiModel.MovieItem -> R.layout.movie_list_item
+            is MovieFeedUiModel.AdvertisementItem -> R.layout.ad_banner_item
+        }
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int,
+    ): ViewHolder {
+        return when (viewType) {
+            R.layout.movie_list_item -> {
+                MovieViewHolder(parent, onReserveClick)
+            }
+
+            R.layout.ad_banner_item -> {
+                AdViewHolder(parent)
+            }
+            else -> throw IllegalArgumentException(INVALID_VIEW_TYPE)
+        }
```

Comment:
```suggestion
        return when (viewType) {
            R.layout.movie_list_item -> MovieViewHolder(parent, onReserveClick)
            R.layout.ad_banner_item -> AdViewHolder(parent)
            else -> throw IllegalArgumentException(INVALID_VIEW_TYPE)
        }
```

이렇게 간단하게 작성할 수도 있겠습니다.

Reply:
[해당 내용 관련 커밋입니다](https://github.com/woowacourse/android-movie-theater/pull/100/commits/807599499e8beb534a299478c4ac001eca06697f)